gobblin-core/src/main/java/gobblin/source/extractor/extract/sftp/SftpFsHelper.java

```diff
@@ -121,7 +165,7 @@ public InputStream getFileStream(String file)
       throws FileBasedHelperException {
     SftpGetMonitor monitor = new SftpGetMonitor();
     try {
-      return this.channelSftp.get(file, monitor);
+      return getSftpChannel().get(file, monitor);
```

Comment:
Do you know if the channel gets auto-closed if someone closes the inputstream? My guess is probably not. In that case, I suggest you return a wrapper InputStream that will do it since you have a dedicated channel per file. An even better solution is to have a pool of InputStreams that can be reused. My guess is that creating a new channel is an expensive operation because you need to have PK authentication.